### PR TITLE
Update package builder tester

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
This has updates to PackageBuilderTests API

### CODE REVIEW

File: Tests/PackageLoading/ConventionTests.swift [hunk 1]
```diff
@@ -159,21 +158,42 @@ private extension FileSystem {
     }
 }
 
-/// Loads a package using PackageBuilder at the given path. If package description is provided it is used,
-/// otherwise a default package with name of directory is passed to manifest.
+/// A convenience enum for package builder tester. Since most of our convention tests
+/// only requires a Package object with the name filled in.
+enum PackageOrName {
```
Comment (aciidb0mb3r) [inline Tests/PackageLoading/ConventionTests.swift]: @ddunbar @abertelrud I am open to suggestion for better name here

File: Tests/PackageLoading/ConventionTests.swift [hunk 2]
```diff
@@ -277,11 +276,10 @@ final class PackageBuilderTester {
...
-        }
-
-        func checkSources(_ paths: String...) {
+        func checkSources(root: String? = nil, paths: String...) {
+            if let root = root {
+                XCTAssertEqual(module.sources.root, AbsolutePath(root))
+            }
             for path in paths {
                 XCTAssert(sources.contains(RelativePath(path)), "\(path) not found in module \(module.name)")
             }
```
Comment (ddunbar) [inline Tests/PackageLoading/ConventionTests.swift]: Shouldn't this check that sources contains *exactly* paths? Right now extra sources are not an error.

Comment (aciidb0mb3r) [inline Tests/PackageLoading/ConventionTests.swift]: How do you think that should be handled? check at exit or clients made required to pass in all the paths at once?

File: Tests/PackageLoading/ConventionTests.swift (hunk 1 above)
Comment (ddunbar) [inline Tests/PackageLoading/ConventionTests.swift]: How often do we use the variant when a `Package` is provided? Maybe we should do that using another mechanism (we could just do it by including an appropriate `Package.swift` in the FS)?

File: Tests/PackageLoading/ConventionTests.swift (hunk 2 above)
Comment (ddunbar) [inline Tests/PackageLoading/ConventionTests.swift]: Yeah, I think so. We can have an init that takes a `[String]` if there are tests that need to factor out the list construction across separate lines.

File: Tests/PackageLoading/ConventionTests.swift (hunk 1 above)
Comment (aciidb0mb3r) [inline Tests/PackageLoading/ConventionTests.swift]: hmm I guess we only need it when we need to check for excluded paths and target dependencies

Comment (ddunbar) [inline Tests/PackageLoading/ConventionTests.swift]: It does seem convenient to be able to pass in an explicit one sometimes...

File: Tests/PackageLoading/ConventionTests.swift [hunk 3]
```diff
@@ -158,21 +158,42 @@ private extension FileSystem {
...
 ///
 /// - Parameters:
+///     - packageOrName: Package name or PackageDescription instance to use for loading this package.
 ///     - path: Directory where the package is located.
 ///     - in: FileSystem in which the package should be loaded from.
-///     - package: PackageDescription instance to use for loading this package.
 ///
 /// - Throws: ModuleError, ProductError
-private func loadPackage(_ path: AbsolutePath, in fs: FileSystem, package: PackageDescription.Package? = nil) throws -> PackageModel.Package {
+private func loadPackage(_ packageOrName: PackageOrName, path: AbsolutePath, in fs: FileSystem) throws -> PackageModel.Package {
```
Comment (ddunbar) [inline Tests/PackageLoading/ConventionTests.swift]: What if loadPackage took in the `PackageDescription.Package`, and there were just two `init` functions, one that takes a name and one that takes a `PackageDescription.Package`, and the former just makes a dummy package and calls the other. That avoids needing the extra type.

Comment (aciidb0mb3r) [inline Tests/PackageLoading/ConventionTests.swift]: Yeah that is better, will update

**Comment by ddunbar** [comment]
LGTM!

**Pull request merged**